app/screens/permalink/permalink.js

```diff
@@ -0,0 +1,458 @@
+import PostListRetry from 'app/components/post_list_retry';
+import SafeAreaView from 'app/components/safe_area_view';
+import {wrapWithPreventDoubleTap} from 'app/utils/tap';
+import {changeOpacity, makeStyleSheetFromTheme} from 'app/utils/theme';
+
+Animatable.initializeRegistryWithDefinitions({
+    growOut: {
+        from: {
+            opacity: 1,
+            scale: 1
+        },
+        0.5: {
+            opacity: 1,
+            scale: 3
+        },
+        to: {
+            opacity: 0,
+            scale: 5
+        }
+    }
+});
+
+export default class Permalink extends PureComponent {
+    static propTypes = {
+        actions: PropTypes.shape({
+            getPostsAfter: PropTypes.func.isRequired,
+            getPostsBefore: PropTypes.func.isRequired,
+            getPostThread: PropTypes.func.isRequired,
+            getChannel: PropTypes.func.isRequired,
+            handleSelectChannel: PropTypes.func.isRequired,
+            handleTeamChange: PropTypes.func.isRequired,
+            joinChannel: PropTypes.func.isRequired,
+            loadThreadIfNecessary: PropTypes.func.isRequired,
+            markChannelAsRead: PropTypes.func.isRequired,
+            markChannelAsViewed: PropTypes.func.isRequired,
+            selectPost: PropTypes.func.isRequired,
+            setChannelDisplayName: PropTypes.func.isRequired,
+            setChannelLoading: PropTypes.func.isRequired
+        }).isRequired,
+        channelId: PropTypes.string,
+        channelName: PropTypes.string,
+        channelTeamId: PropTypes.string,
+        currentTeamId: PropTypes.string.isRequired,
+        currentUserId: PropTypes.string.isRequired,
+        focusedPostId: PropTypes.string.isRequired,
+        isPermalink: PropTypes.bool,
+        myMembers: PropTypes.object.isRequired,
+        navigator: PropTypes.object,
+        onClose: PropTypes.func,
+        onPermalinkPress: PropTypes.func,
+        onPress: PropTypes.func,
+        postIds: PropTypes.array,
+        theme: PropTypes.object.isRequired
+    };
+
+    static defaultProps = {
+        onPress: () => true,
+        postIds: []
+    };
+
+    static contextTypes = {
+        intl: intlShape.isRequired
+    };
+
+    constructor(props, context) {
+        super(props);
+
+        const {postIds, channelName} = props;
+        let show = false;
+
+        if (postIds && postIds.length >= 10) {
+            show = true;
+        }
+
+        if (!show) {
+            this.loadPosts(props, context);
+        }
+
+        this.state = {
+            title: channelName,
+            show,
+            error: '',
+            retry: false
+        };
+    }
+
+    componentWillReceiveProps(nextProps) {
+        if (this.props.channelName !== nextProps.channelName) {
+            this.setState({title: nextProps.channelName});
+        }
+
+        if (this.props.focusedPostId !== nextProps.focusedPostId) {
+            this.setState({show: false}, () => {
```

Comment:
Maybe we should only change `this.state.show` if `nextProps.postIds && nextProps.postIds.length < 10`? I don't follow why we'd show this as loading when changing focused posts if we don't need to